### PR TITLE
Cov inf

### PR DESCRIPTION
**Pull request by agramfort**
follow up on #1637

@Eric89GXL problem should be solved.

please give it a try.

### CODE REVIEW

File: mne/cov.py [hunk 1]
```diff
@@ -760,10 +759,32 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
     return out
 
 
+def _logdet(A):
+    vals = linalg.eigh(A)[0]
+    vals = np.abs(vals)  # avoid negative values (numerical errors)
```
Comment (agramfort) [inline mne/cov.py]: @Eric89GXL that's the culprit. eigh returns negative eig vals on a rank deficient symmetric matrix.

Comment (dengemann) [inline mne/cov.py]: add at least a pseudo doc string please

File: mne/cov.py [hunk 2]
```diff
@@ -760,10 +759,32 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
...
+    return np.sum(np.log(vals))
+
+
+def _gaussian_loglik_scorer(est, X, y=None):
```
Comment (dengemann) [inline mne/cov.py]: same here

File: mne/cov.py [hunk 3]
```diff
@@ -760,10 +759,32 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
...
+    # compute empirical covariance of the test set
+    precision = est.get_precision()
+    n_samples, n_features = X.shape
+    log_like = np.zeros(n_samples)
+    log_like = -.5 * (X * (np.dot(X, precision))).sum(axis=1)
+    log_like -= .5 * (n_features * log(2. * np.pi)
+                      - _logdet(precision))
+    out = np.mean(log_like)
+    if np.isinf(out):
+        import ipdb; ipdb.set_trace()
```
Comment (dengemann) [inline mne/cov.py]: remore this please

**Comment by dengemann** [comment]
besides my comments looks good!

**Comment by agramfort** [comment]
done

**Comment by dengemann** [comment]
Can you add a test for this fix?

**Comment by agramfort** [comment]
your tests still pass so I assume I did not break anything :)

**Comment by dengemann** [comment]
> your tests still pass so I assume I did not break anything :)

Hum, this means that Eric should probably rerun his example script. (Cc @Eric89GXL)

**Comment by agramfort** [comment]
yeah let's give some homework to @Eric89GXL ;)

**Comment by Eric89GXL** [comment]
LGTM, I'll test it on my script tomorrow then merge if I get 0% infs :)

But first you need to fix your style:

https://travis-ci.org/mne-tools/mne-python/jobs/51649710#L795

**Comment by agramfort** [comment]
style fixed.

**Comment by Eric89GXL** [comment]
Testing now

**Comment by Eric89GXL** [comment]
It worked, no more infs! Thanks @agramfort

**Comment by agramfort** [comment]
yes ! :beers: for me :)

**Comment by dengemann** [comment]
Thanks @agramfort you deseve a few German Hefeweizen for this!!!

**Pull request merged**